datalad_container/utils.py

```diff
@@ -9,3 +15,62 @@ def get_container_command():
         raise RuntimeError("Did not find apptainer or singularity")
 
 
+def get_container_configuration(
+    ds: Dataset,
+    name: str | None = None,
+) -> dict:
+    """Report all container-related configuration in a dataset
+
+    Such configuration is identified by the item name pattern::
+
+      datalad.containers.<container-name>.<item-name>
+
+    Parameters
+    ----------
+    ds: Dataset
+      Dataset instance to report configuration on.
+    name: str, optional
+      If given, the reported configuration will be limited to the container
+      with this exact name. In this case, only a single ``dict`` is returned,
+      not nested dictionaries.
```

Comment:
could we make it more explicit, i.e. separate out a `get_containers_configuration(ds)` ?  internally `get_container_configuration(ds, name)` can just call that `get_containers_configuration(ds)`, I don't care.  But having singular semantic in the name but getting a dict for multiple containers will eventually confuse someone like me at some point because e.g. I thought without a name specified it would return the config for the "only" container (as e.g. `containers-run` behaves IIRC) and raise KeyError or smth like that if multiple are found -- so it becomes ambiguous.

Reply:
Just to clarify. Your response to the description in https://github.com/datalad/datalad-container/pull/242/commits/dbb2f914e43cf467858a2b7941ded686ae88908f (commit message) is: Let's nevertheless always read and post-process all configuration, make all filesystem checks for all containers, and then throw away all that information, but for one container.

Rather than that, you can also get that semantic cleanliness by having two near-copies of that function.

Just to be clear, I am not here to argue -- I just needs this merged, and I don't care in which way. I just do not want to guess what should be done and avoid future iterations on this code.

Reply:
I have missed that there is a `continue` shortcut, so indeed some code is skipped... even though complexity stays `O(N)`, there would indeed be not only a waste of cycles but also possibly errors on unrelated containers which should be avoidable. I guess a way could be to make current function `_private` and then expose it via two clean interfaces calling it out with or without a name.
I will leave it up to you whether to implement that or proceed with current version.

Reply:
This is a nice suggestion! I will change it to use a private helper and two clean interfaces.

Depending on your reviewing capabilities, I might do this in a separate follow-up PR, given that more PR are built on this, and they would all need an update.

Reply:
Then let's merge